agent/engine/execcmd/manager_init_task_windows.go

```diff
@@ -16,18 +16,214 @@
-// Note: exec cmd agent is a linux feature, thus implemented here as a no-op.
-func (m *manager) InitializeContainer(taskId string, container *apicontainer.Container, hostConfig *dockercontainer.HostConfig) error {
+var (
+	ecsAgentExecDepsDir = config.AmazonECSProgramFiles + "\\managed-agents\\execute-command"
+
+	// ecsAgentDepsBinDir is the directory where ECS Agent will read versions of SSM agent
+	ecsAgentDepsBinDir = ecsAgentExecDepsDir + "\\bin"
+
+	containerDepsFolder = "C:\\Program Files\\Amazon\\SSM"
+
+	SSMAgentBinName       = "amazon-ssm-agent.exe"
+	SSMAgentWorkerBinName = "ssm-agent-worker.exe"
+	SessionWorkerBinName  = "ssm-session-worker.exe"
+
+	HostLogDir      = config.AmazonECSProgramData + "\\exec"
+	ContainerLogDir = config.AmazonProgramData + "\\SSM"
+
+	SSMPluginDir = "C:\\Program Files\\Amazon\\SSM\\Plugins"
+	// since ecs agent windows is not running in a container, the agent and host log dirs are the same
+	ECSAgentExecLogDir = config.AmazonECSProgramData + "\\exec"
+
+	// ECSAgentExecConfigDir is the directory where ECS Agent will write the ExecAgent config files to
+	ECSAgentExecConfigDir = ecsAgentExecDepsDir + "\\" + ContainerConfigDirName
+	// HostExecConfigDir is the dir where ExecAgents Config files will live
+	HostExecConfigDir = hostExecDepsDir + "\\" + ContainerConfigDirName
+
+	configFiles = []string{
+		"amazon-ssm-agent.json",
+		"seelog.xml",
+	}
+
+	execAgentConfigTemplate = `{
+"Mgs": {
+	"Region": "",
+	"Endpoint": "",
+	"StopTimeoutMillis": 20000,
+	"SessionWorkersLimit": %d
+},
+"Agent": {
+	"Region": "",
+	"OrchestrationRootDir": "",
+	"ContainerMode": true
+}
+}`
+
+	execAgentLogConfigTemplate = `<!--amazon-ssm-agent uses seelog logging -->
+<!--Seelog has github wiki pages, which contain detailed how-tos references: https://github.com/cihub/seelog/wiki -->
+<!--Seelog examples can be found here: https://github.com/cihub/seelog-examples -->
+<seelog type="adaptive" mininterval="2000000" maxinterval="100000000" critmsgcount="500" minlevel="info">
+    <exceptions>
+        <exception filepattern="test*" minlevel="error"/>
+    </exceptions>
+    <outputs formatid="fmtinfo">
+        <console formatid="fmtinfo"/>
+        <rollingfile type="size" filename="C:\ProgramData\Amazon\SSM\Logs\amazon-ssm-agent.log" maxsize="30000000" maxrolls="5"/>
+        <filter levels="error,critical" formatid="fmterror">
+            <rollingfile type="size" filename="C:\ProgramData\Amazon\SSM\Logs\errors.log" maxsize="10000000" maxrolls="5"/>
+        </filter>
+    </outputs>
+    <formats>
+        <format id="fmterror" format="%Date %Time %LEVEL [%FuncShort @ %File.%Line] %Msg%n"/>
+        <format id="fmtdebug" format="%Date %Time %LEVEL [%FuncShort @ %File.%Line] %Msg%n"/>
+        <format id="fmtinfo" format="%Date %Time %LEVEL %Msg%n"/>
+    </formats>
+</seelog>`
+)
+
+func (m *manager) InitializeContainer(taskId string, container *apicontainer.Container, hostConfig *dockercontainer.HostConfig) (rErr error) {
```

Comment:
I'd prefer not to fork this whole method as it has many commonalities with linux. Instead we should probably just add a sub-method called `addRequiredBindMounts` or something similar. That way we can preserve a single method (i.e. reuse as much as possible) while still being able to configure windows in a windows-specific way

Reply:
will refactor and push